packages/autorest.python/autorest/codegen/__init__.py

```diff
@@ -164,6 +164,7 @@ def _build_code_model_options(self) -> Dict[str, Any]:
                 low_level_client or version_tolerant,
             ),
             "generate_sample": self.options.get("generate-sample", False),
+            "default_api_version": self.options.get("default-api-version"),
```

Comment:
@iscai-msft According to our [discussion](https://github.com/Azure/autorest.python/pull/1505#discussion_r996214682), we don't want users to use multiapi package with specific api-version, so we should only generate samples for default api-version. However, we can't get default api-version until all the api-versions are generated which is too late. I think we could add flag to tell codegen which api-version to generate samples.

Reply:
I believe by generating at this step of the process, you will still be generating samples within an api version subfolder, even though it's the default api version sub folder, and it shouldn't be on that level, it should be one-up.

What is the timeline for adding multiapi support? The more I think about it, I think we should have sample serialization as a separate plugin. For single-api version, it will run after generation, and for multi-api version, it will run after the multiapiscript. What are your thoughts?

Reply:
@iscai-msft The main target of genereated_samples is to show SDK sample in doc UI like: [CDN](https://learn.microsoft.com/en-us/rest/api/cdn/custom-domains/create?tabs=Python#examples). And the doc UI only show latest stable api-version from swagger repo, so we just need to confirm the latest stable api-version has samples. And default api-version of multiapi package is exactly latest stable api-version which meets requirements. I think your thought is better solution but it costs more effort and may not have better effect. 
**In view of the effort and target, I think current solution meets the requirements.**

Reply:
@msyyc we don't want people to ever go into the individual API version folders. We want to completely handle multiple api versions for users in our multiapi packages so they never go into the individual api version folders. Right now, let's say we have a multiapi package with default API version `2022-10-01`, we will generate the samples in the folder `azure/my/sdk/v2022_10_01/generated_samples`. However, we want the samples to be in `azure/my/sdk/generated_samples`.

I'm ok with doing something a little hacky for now to move the generated samples to the root of the sdk because of your target concerns, but we should have an issue to fix the implementation for this down the road.

Reply:
@iscai-msft Agreed and I create https://github.com/Azure/autorest.python/issues/1709 to track for better solution in the future. Could you help merge the PR and release a new version? We need to reach 100% for sample coverage.

Reply:
I can't until you can confirm that the samples are being generated at the root of the sdk. I'm ok with us doing something hacky to make sure they're being generated at `azure/my/sdk/generated_samples` instead of `azure/my/sdk/v2022_01_30/generated_samples` for now, but we need to generate at the root before merging because we can't point people to version specific subfolders

Reply:
@iscai-msft Sure. Here is the [demo](https://github.com/Azure/azure-rest-api-specs/tree/multiapi-generated-samples/specification/redhatopenshift/resource-manager/redhatopenshift/azure-mgmt-redhatopenshift/generated_samples) that we can see the `generated_samples` is not in version subfolder but in root folder.